adlink_camera_sync.py

```diff
@@ -0,0 +1,252 @@
+            os.popen("i2cset -f -y 2 0x66 0x04 0xff").readlines
+            print("Set to frame Sync mode")
+        except ValueError as e:
+            print(e)
+
+        # Start the daemon
+        self.daemonize()
+        self.camera.run()
+
+    def stop(self):
+        """Stop the daemon."""
+
+        # Get the pid from the pidfile
+        try:
+            with open(self.pidfile, 'r') as pf:
+                pid = int(pf.read().strip())
+        except IOError:
+            pid = None
+
+        if not pid:
+            message = "pidfile {0} does not exist. " + \
+                "Daemon not running?\n"
+            sys.stderr.write(message.format(self.pidfile))
+            return  # not an error in a restart
+
+        # Try killing the daemon process
+        try:
+            while 1:
+                os.kill(pid, signal.SIGTERM)
+                time.sleep(0.1)
+        except OSError as err:
+            e = str(err.args)
+            if e.find("No such process") > 0:
+                if os.path.exists(self.pidfile):
+                    os.remove(self.pidfile)
+            else:
+                print(str(err.args))
+                sys.exit(1)
+
+        # Set the camera to free mode
+        try:
+            os.popen("echo 0 > /sys/module/leopard_ar0233/parameters/trigger_mode").readlines
+            os.popen("i2cset -f -y 2 0x66 0x04 0xf0").readlines
+            print("Set to free mode")
+        except ValueError as e:
+            print(e)
+
+    def restart(self):
+        """Restart the daemon."""
+
+        self.stop()
+        self.start()
+
+
+if __name__ == "__main__":
+    # Argument definition
+    # ./adlink_camera_sync [mode] [-f freq] [-t gpio_num]
+    # * mode:
+    #   - sync:
+    #     1. set frame sync mode
+    #     2. enable isr if gpio_num != None
+    #     3. enable trigger daemon
+    #   - free:
+    #     1. disable trigger daemon
+    #     2. set free run mode
+    # * While sync mode, supports two options
+    #   -f freq: the trigger frequency (default 5 Hz)
+    #   -t gpio_num: whether triggered by GPIO in
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('mode', type=str, choices=[
+                        'sync', 'free', 'restart'], help='Camera trigger mode: frame sync / free run mode')
+    parser.add_argument('-f', '--freq', type=int, default='5',  choices=range(1, 21), 
+                        metavar="[1~20]", help='The camera trigger frequency (default 5 Hz)')
+    parser.add_argument('-t', '--gpio_num', type=int, default=None,
+                        required=False, help='Whether triggered by GPIO in (default None)')
+    args = parser.parse_args()
+    
+    # set daemon : pidfile, gpio_pin(isr), hz
+    hz = args.freq
+    gpio_pin = args.gpio_num
+    camera = Camera(hz, gpio_pin)
+    MyDaemon = daemon('/tmp/daemon-example.pid', camera)
+
+    if 'sync' == args.mode:
+        MyDaemon.start()
```

Comment:
Maybe you don't need to name it as start/stop.
Just use `run_sync()` and `run_free()`

Reply:
OK, I will change the function name. Your is more clear.